src/lib/components/ng-http-loader.component.ts

```diff
@@ -20,7 +20,7 @@ import { Spinkit } from '../spinkits';
     styleUrls: ['./ng-http-loader.component.scss']
 })
 export class NgHttpLoaderComponent implements OnDestroy, OnInit {
-    public isSpinnerVisible: boolean;
+    public isSpinnerVisibleSubject = new BehaviorSubject<boolean>(false);
```

Comment:
@Akkenar I would like the subject to be private :( But I do things like that in tests :

`component.isSpinnerVisibleSubject = new BehaviorSubject<boolean>(true);`  I'd prefer adding a setter for the observable.

Reply:
I'm not overly worried about the `isSpinnerVisibleSubject` not being private as it's only public within your component. As a consumer of the library, I'd use the component in templates and thus I care very little about this.

Reply:
Agreed.

So the another way would be to have more of an integration test with the assert checking the resulting DOM rather than the value of the component. But that's cumbersome.

I'm still in the process of figuring what the f*** we're suppose to do with tests (unit or integration) in Angular.

Reply:
Yes I don't want to do that. Testing angular is very painful per se. :/ And the test suite is here is yet very large for such a small component.

Reply:
Feels weird to have to mock on such as small scale, but it does help indeed!

Reply:
As long as it avoids mutability, it's good enough for me :D

Reply:
Sounds fair! Let's merge your change then.

Reply:
Will polish a little bit more and publish a first beta later. Thanks for your help !